### DISCUSSION TITLE
Path2D is not Defined

### DISCUSSION THREAD

**Discussion by iamstevedavis** [gatsbyjs/gatsby#31170] [Help] [answered]
Hey all,

I am trying to use Path2D. It works on `develop` but of course on `build` I get Path2D is undefined. I found this answer https://github.com/sveltejs/sapper-legacy.svelte.dev/pull/45/files but I can't seem to get it to work. Here is basically my code:

```
const someSVG = '...';

const draw = (ctx, location) => {
  const SVG_PATH = new Path2D(someSVG);
  ...
  ctx.fill(SVG_PATH);
  ...
};

const MyCanvas = () => {
  ...
  do something with draw()
  ...
};
  
export default MyCanvas;
```
hoping someone can help me out, thanks all! :+1:

**Comment by herecydev** [comment, marked as answer]
You could bail out early if `window` is not defined. This would allow SSR to work

**Reply by iamstevedavis** [reply]
<3 100% worked. 
```
if (typeof window !== 'undefined') {
  SVG_PATH = new Path2D(mySVG);
}
```
